### PR TITLE
Initial build changes for 8.2

### PR DESCRIPTION
**Pull request by bduranleau-nr**
Overview:
- cc3c88e5eb2e8b0a2fd85494fea22fe71b9220a1 updates Docker to pull the 8.2 image
- 026e0ba305a70f2ef00639b776c1ca5084e6893a adds the Zend API No
- 3946ab3f0f4ab6d32e019db426e714972cc04cd9 addresses the removal of the [num_additional_modules](https://github.com/php/php-src/blob/PHP-8.2/UPGRADING.INTERNALS#L109)
- 9cb040aab11a9055b032075e39edb1fb5b596396 addresses the addition of a `type` to [REGISTER/UNREGISTER_INI_ENTRIES](https://github.com/php/php-src/blob/PHP-8.2/Zend/zend_ini.h#L199)
- dabca0db641e83c859e0e2fbc8f2bf5504811ddf updates GHA workflow build considerations

### CODE REVIEW

**Comment by bduranleau-nr** [comment]
ok jenkins

File: .github/workflows/agent-build.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
     strategy:
       matrix:
         os: [linux]
-        php_ver: ['7.0', '7.1-zts', '7.2', '7.3', '7.4', '8.0', '8.0-zts', '8.1', '8.1-zts']
+        php_ver: ['7.0', '7.1-zts', '7.2', '7.3', '7.4', '8.0', '8.0-zts', '8.1', '8.1-zts', '8.2']
         arch: ['x64', 'x86']
```
Comment (mfulb) [inline .github/workflows/agent-build.yml:42]: You can just remove 'x86' from this and probably lose all the `excludes`.

**Review by ZNeumann — APPROVED** [review]

File: .github/workflows/agent-build.yml (hunk 1 above)
Comment (bduranleau-nr) [reply .github/workflows/agent-build.yml:42]: Agreed, but probably best to handle that in a separate PR

**Comment by bduranleau-nr** [comment]
ok jenkins

File: .github/workflows/agent-build.yml (hunk 1 above)
Comment (mfulb) [reply .github/workflows/agent-build.yml:42]: I've done it in the oapi branch already so it will get picked up.

**Review by mfulb — APPROVED** [review]

**Pull request merged**